mainwindow.cpp

```diff
@@ -209,31 +220,7 @@ void MainWindow::translate(QString const &text) {
  */
 
 void MainWindow::resetTranslator(QString dirname) {
-    // Disconnect existing slots:
-    if (translator_) {
-        disconnect(translator_.get());
-    }
-    QString model0_path = dirname + "/";
-    ui_->localModels->setEnabled(false); // Disable changing the model while changing the model
-    ui_->translateAction->setEnabled(false); //Disable the translate button before the swap
-    ui_->translateButton->setEnabled(false);
-
-    translator_.reset(); // Do this first to free the object.
-    translator_.reset(new MarianInterface(model0_path, models_.getSettings() , this));
-
-    ui_->translateAction->setEnabled(true); // Reenable it
-    ui_->localModels->setEnabled(true); // Disable changing the model while changing the model
-    ui_->translateButton->setEnabled(true);
-
-    // Set up the connection to the translator
-    connect(translator_.get(), &MarianInterface::pendingChanged, ui_->pendingIndicator, &QProgressBar::setVisible);
-    connect(translator_.get(), &MarianInterface::translationReady, this, [&](QString translation) {
-        ui_->outputBox->setText(translation);
-        ui_->localModels->setEnabled(true); // Re-enable model changing
-        ui_->translateAction->setEnabled(true); // Re-enable button after translation is done
-        ui_->translateButton->setEnabled(true);
-    });
-
+    translator_->setModel(dirname + "/", models_.getSettings());
     translate();
```

Comment:
We always translate after changing the model?

Reply:
Yes, to show that the new model has been loaded and have the translation show up in the new language. If the current input is empty string it will just sail right through without much overhead.